File: backend-csharp/Models/GeneratedRecipe.cs
// Models/GeneratedRecipe.cs

public class GeneratedRecipe
{
    public int Id { get; set; }
    // 对应 output_schema 里的 menus -> recipes
    public string Title { get; set; }
    public string ShortDescription { get; set; }
    public int Servings { get; set; }
    public int CookingTimeMin { get; set; }
    public string Difficulty { get; set; } // easy, medium, hard
    public double TotalCaloriesEstimate { get; set; }
    
    // 步骤 (存 JSONB)
    // 对应 Python output: steps array
    public string StepsJson { get; set; } 
    
    // 食材 (存 JSONB 或者关联表)
    // 对应 Python output: ingredients array
    public string IngredientsJson { get; set; } 
    
    public DateTime GeneratedAt { get; set; } = DateTime.Now;
}

File: backend-java/src/main/java/com/souschef/entity/GeneratedRecipe.java
package com.souschef.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "generated_recipes")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class GeneratedRecipe {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;
    
    @Column(name = "title", nullable = false)
    private String title;
    
    @Column(name = "short_description", columnDefinition = "TEXT")
    private String shortDescription;
    
    @Column(name = "servings", nullable = false)
    private Integer servings;
    
    @Column(name = "cooking_time_min", nullable = false)
    private Integer cookingTimeMin;
    
    @Column(name = "difficulty", nullable = false)
    private String difficulty;
    
    @Column(name = "total_calories_estimate", nullable = false)
    private Double totalCaloriesEstimate;
    
    @Column(name = "used_cookwares_json", columnDefinition = "TEXT")
    private String usedCookwaresJson = "[]";
    
    @Column(name = "steps_json", columnDefinition = "TEXT")
    private String stepsJson = "[]";
    
    @Column(name = "ingredients_json", columnDefinition = "TEXT")
    private String ingredientsJson = "[]";
    
    @Column(name = "generated_at", nullable = false)
    private LocalDateTime generatedAt = LocalDateTime.now();
}


